apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-library/src/main/java/gov/cms/model/dsl/codegen/library/DataTransformer.java

```diff
@@ -0,0 +1,849 @@
+   * @return true if the value is non-null
+   */
+  private boolean nonNull(String fieldName, Object value, boolean nullable) {
+    if (value != null) {
+      return true;
+    }
+    if (!nullable) {
+      addError(fieldName, "is null");
+    }
+    return false;
+  }
+
+  /**
+   * Used internally to stop transformation if the value length is out of bounds.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param value non-null string value
+   * @param minLength minimum valid length for the string
+   * @param maxLength maximum valid length for the string
+   * @return true if the value has a valid length
+   */
+  private boolean lengthOk(String fieldName, String value, int minLength, int maxLength) {
+    final int length = value.length();
+    if (length < minLength || length > maxLength) {
+      addError(
+          fieldName, "invalid length: expected=[%d,%d] actual=%d", minLength, maxLength, length);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Detects if the two values are the same. Adds an appropriate error message if they are not.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param expectedValue expected value
+   * @param actualValue actual value of the field
+   * @return true if the values are equal
+   */
+  private boolean valueMatches(String fieldName, String expectedValue, String actualValue) {
+    final boolean matches =
+        (expectedValue == null && actualValue == null)
+            || (expectedValue != null && expectedValue.equals(actualValue));
+    if (!matches) {
+      final String masked = maskString(actualValue, expectedValue);
+      addError(fieldName, "value mismatch: masked=%s", masked);
+    }
+    return matches;
+  }
+
+  /**
+   * Adds a formatted error message to the list of errors.
+   *
+   * @param fieldName name of the field containing the error
+   * @param errorFormat {@link String#format} compatible format string for the message
+   * @param args array of arguments to fill placeholders in the format string
+   */
+  public void addError(String fieldName, String errorFormat, Object... args) {
+    final String message = String.format(errorFormat, args);
+    errors.add(new ErrorMessage(fieldName, message));
+  }
+
+  /**
+   * Convert a {@link Timestamp} into an {@link Instant}.
+   *
+   * @param timestamp the {@link Timestamp} to convert
+   * @return the corresponding {@link Instant}
+   */
+  public Instant instant(Timestamp timestamp) {
+    return Instant.ofEpochSecond(timestamp.getSeconds(), timestamp.getNanos());
+  }
+
+  /**
+   * Produces a string suitable for logging comparison mis-matches of two potentially sensitive
+   * strings. The resulting string contains '.' to indicate characters that match, '+' to indicate
+   * an extra character at the end of the string, '-' to indicate a missing character at the end of
+   * the string, '#' to indicate a mismatching character.
```

Comment:
I'm not sure where exactly, but it seems like this valuable information should also be present somewhere less buried, perhaps in the top level README or on the public getErrors() method?